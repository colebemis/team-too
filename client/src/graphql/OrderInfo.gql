query orderInfo($id: OrderWhereUniqueInput!) {
  order(where: $id ) {
    id
    createdAt
    status
    customer {
      name
      email
    }
    billingAddress {
      name
      line1
      line2
      city
      state
      zip
    }
    shippingAddress {
      name
      line1
      line2
      city
      state
      zip
    }
    products {
      id
      title
      quantity
      price
    }
    payment {
      name
      number
      expDate
    }
    subtotal
    tax
    total
  }
}
