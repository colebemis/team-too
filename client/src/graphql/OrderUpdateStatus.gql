mutation orderUpdateStatus ($id : ID!, $status : String!) {
  updateOrder (where: { id: $id }, data: { status : $status }) {
    id
    status
  }
}