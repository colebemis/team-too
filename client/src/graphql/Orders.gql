query orders {
  orders {
    id
    status
    createdAt
    customer {
      name
    }
    shippingAddress {
      id
    }
  }
}