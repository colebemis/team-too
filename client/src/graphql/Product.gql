query product($id: ID!) {
    product(where: { id: $id }) {
        id
        title
        imageURL
        description
        price
        stock
    }
}