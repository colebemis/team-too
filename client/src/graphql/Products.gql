query products {
    products {
        id
        description
        title
        price
        imageURL
        stock
        updatedAt
        categories {
        name
        }
    }
}