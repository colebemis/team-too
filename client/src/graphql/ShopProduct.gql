query
{
    products
    {
        id
        info
        name
        price
        image
    }
}