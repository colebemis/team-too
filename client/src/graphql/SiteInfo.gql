query SiteInfo {
  siteInfo {
    id
    phone
    email
    address {
      line1
      line2
      city
      state
      zip
    }
    hours {
      id
      index
      day
      open
      close
    }
    about
    services
  }
}
