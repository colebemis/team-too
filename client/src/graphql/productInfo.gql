
query productInfo($id: ProductWhereUniqueInput!) {
    product(where: $id)
    {
        title
        imageURL
        description
        price

    }
}
