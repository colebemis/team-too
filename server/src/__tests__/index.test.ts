import { gql } from "apollo-server";
import { graphql } from "graphql";
import { importSchema } from "graphql-import";
import { addMockFunctionsToSchema, makeExecutableSchema } from "graphql-tools";
import * as path from "path";

// Set up and mock graphql server
const typeDefs = gql(
  importSchema(path.resolve(__dirname, "../schema.graphql")),
);
const schema = makeExecutableSchema({ typeDefs });
addMockFunctionsToSchema({ schema });

it("returns user name and email", async () => {
  const query = `
    query  {
      users {
        name,
        email
      }
    }
  `;

  const result = await graphql(schema, query);

  expect(result).toMatchSnapshot();
});

it("returns product title, description, and price", async () => {
  const query = `
    query  {
      products {
        title,
        description,
        price
      }
    }
  `;

  const result = await graphql(schema, query);

  expect(result).toMatchSnapshot();
});
