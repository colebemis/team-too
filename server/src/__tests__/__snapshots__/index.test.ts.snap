// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns product title, description, and price 1`] = `
Object {
  "data": Object {
    "products": Array [
      Object {
        "description": "Hello World",
        "price": -67.38944318055582,
        "title": "Hello World",
      },
      Object {
        "description": "Hello World",
        "price": 68.56515210384657,
        "title": "Hello World",
      },
    ],
  },
}
`;

exports[`returns user name and email 1`] = `
Object {
  "data": Object {
    "users": Array [
      Object {
        "email": "Hello World",
        "name": "Hello World",
      },
      Object {
        "email": "Hello World",
        "name": "Hello World",
      },
    ],
  },
}
`;
